eclass/rakudo.eclass

```diff
@@ -0,0 +1,55 @@
+# Copyright 2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: rakudo.eclass
+# @MAINTAINER:
+# amano.kenji <amano.kenji@proton.me>
+# @BLURB: An eclass for raku modules
+
+EXPORT_FUNCTIONS src_compile src_test src_install
+
+# @ECLASS_VARIABLE: rakudo_test_deps
+# @DESCRIPTION:
+# Packages that shouldn't run tests with dev-raku/App-Prove6
+# @INTERNAL
+declare -A rakudo_test_deps
+rakudo_test_deps[dev-raku/App-Prove6]=1
+rakudo_test_deps[dev-raku/Getopt-Long]=1
+rakudo_test_deps[dev-raku/TAP]=1
+
+if [[ ${rakudo_test_deps[${CATEGORY}/${PN}]} ]]; then
```

Comment:
This would really be much more readable as `case ${CATEGORY}/${PN} in` and setting some meaningfully named variable. Also, declare `BDEPEND` first, then perhaps just conditionally append to it? Also, `:=` is meaningless in `BDEPEND`, it's allowed only so you could do `BDEPEND=${RDEPEND}`.

Also, do these test deps really have no tests, or are you skipping them to avoid circular deps? If the latter, that's wrong.

Reply:
The test deps run their own tests without prove6.

Reply:
Can you show me how you would do it? I don't know much about bash.

Reply:
I just did it in my own way. See whether you like it.

Reply:
Resolving it for now.